### PR TITLE
[py-mdanalysis] new version and added dependencies

### PR DESCRIPTION
**Pull request by qwertos**
cc: @aweits 

Original commit message:

Author: Andrew Elble <aweits@rit.edu>
Date:   Thu Nov 14 08:35:14 2019 -0500

    mdanalysis

### CODE REVIEW

**Comment by scheibelp** [comment]
The unit tests are failing because some of the mentioned dependencies don't yet exist in Spack. Could you add pointers to them here (assuming there are PRs for them)?

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 1]
```diff
@@ -38,7 +39,10 @@ class PyMdanalysis(PythonPackage):
     depends_on('py-six@1.4.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: For `python`, can you add type=build/run?

For `numpy`, it looks like 1.13.3 is required now.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 2]
```diff
@@ -38,7 +39,10 @@ class PyMdanalysis(PythonPackage):
...
     depends_on('py-biopython@1.59:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on('py-biopython@1.71:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 3]
```diff
@@ -38,7 +39,10 @@ class PyMdanalysis(PythonPackage):
...
     depends_on('py-networkx@1.0:', type=('build', 'run'))
-    depends_on('py-griddataformats@0.3.2:', type=('build', 'run'))
+    depends_on('py-griddataformats@0.4:', type=('build', 'run'))
+    depends_on('py-gsd', type=('build', 'run'))
+    depends_on('py-mmtf-python', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on('py-mmtf-python@1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 4]
```diff
@@ -38,7 +39,10 @@ class PyMdanalysis(PythonPackage):
...
+    depends_on('py-scikit-learn', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py:43]: Missing dependencies on joblib, scipy, matplotlib, and mock, which are all required, at least in the latest release.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 5]
```diff
@@ -38,7 +39,10 @@ class PyMdanalysis(PythonPackage):
...
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on('py-scikit-learn', when='+analysis', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 6]
```diff
@@ -38,7 +39,10 @@ class PyMdanalysis(PythonPackage):
...
 
     depends_on('py-matplotlib', when='+analysis', type=('build', 'run'))
     depends_on('py-scipy', when='+analysis', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: These are now unconditionally required

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by qwertos** [comment]
Depends on #17200 and #17201

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**